packages/mds-audit/api.ts

```diff
@@ -665,6 +696,59 @@ function api(app: express.Express): express.Express {
     }
   })
 
+  /**
+   * attach media to an audit, uploaded as multipart/form-data
+   */
+  app.post(pathsFor('/trips/:audit_trip_id/attach/:mimetype'), multipartFormUpload, async (req, res) => {
+    const { audit, audit_trip_id } = res.locals
+    if (!audit) {
+      return res.status(404).send({ error: new NotFoundError('audit not found', { audit_trip_id }) })
+    }
+    try {
+      validateFile(req.file)
+    } catch (err) {
+      if (err instanceof ValidationError) {
+        return res.status(400).send({ error: err })
+      }
+      if (err instanceof UnsupportedTypeError) {
+        return res.status(415).send({ error: err })
+      }
+    }
+    const attachment: Attachment | null = await writeAttachment(req.file, audit_trip_id)
```

Comment:
There's a style debate to be had around if this should be able to return `null`, rather than just throwing an error. I prefer try/catch with exceptions as opposed to the way languages like Golang handle error-handling flow. Elsewhere in the code I think we predominantly use try/catch and handle failures as exceptions rather than null returns, so we should definitely try to sync up how we approach it. Anyone wanna chime in?

Reply:
if `writeAttachment` _should_ always succeed, then I would advocate `try/catch`

Reply:
Agreed we should be consistent - and I like letting the caller handle exceptions. Updated.